programs/test/udp_proxy.c

```diff
@@ -681,26 +681,21 @@ int send_delayed()
 }
 
 /*
- * Avoid dropping or delaying a packet that was already dropped twice: this
- * only results in uninteresting timeouts. We can't rely on type to identify
- * packets, since during renegotiation they're all encrypted.  So, rely on
- * size mod 2048 (which is usually just size).
- */
-static unsigned char dropped[2048] = { 0 };
-#define DROP_MAX 2
-
-/* We only drop packets at the level of entire datagrams, not at the level
+ * Avoid dropping or delaying a packet that was already dropped or delayed
```

Comment:
If I understand correctly, given four packets ABCD, the proxy could previously delay A three times and therefore send BCDA, but this is no longer possible. Is this loss of test coverage significant?

Reply:
Good point, but I'm not sure that's how delaying works currently. From my understanding of the code, delaying A means switching from AB to BA but no further. Delaying A multiples times means switching A...AB to BA...A.

I agree that it would probably provide better coverage to allow the proxy to generate orderings like BCAD and BCDA, but I don't think it was doing that before this PR, so allowing the proxy do to that would be an enhancement, independent from the current bug fixing. (To be confirmed, I may mis-remember and mis-read how the code currently works.)

Reply:
Indeed, looking at the code more closely, a packet that's delayed is added to `prev`, and once there it will be sent when `send_delayed` is called and there's no way to re-delay it. So BCDA was never possible, only BACD (A delayed) or CABD (A, B delayed) or DABC (A, B, C delayed) etc.